### ISSUE TITLE
Inconsistent behavior of "session contents restored"

### ISSUE THREAD

**Issue by KostyaTretyak** [microsoft/vscode#142621]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.64.1
Commit: d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03
Date: 2022-02-07T17:17:47.733Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.13.0-28-generic

Steps to Reproduce:

1. Launch two instances of VS Code in different projects.
2. Make `cd ./some/path` in second instance of VS Code.
3. Close second instance of VS Code.
4. Launch second instance of VS Code.
5. See that "session contents restored" not working.

"Session contents restored" works only for last closed instance of VS Code. I think this is unexpected and inconsistent behavior.

**Comment by bpasero** [comment]
Try `window.restoreWindows: preserve`?

**Comment by KostyaTretyak** [comment]
@bpasero , no `window.restoreWindows: all`.

**Comment by KostyaTretyak** [comment]
@bpasero , you still needs more info?

**Comment by bpasero** [comment]
Yes, do you see it too when you configure `window.restoreWindows: preserve`

**Comment by KostyaTretyak** [comment]
Yes, with `window.restoreWindows: preserve` same behavior.

**Comment by bpasero** [comment]
Maybe I misunderstood what `session contents restored` means. What specifically is not restored?

**Comment by KostyaTretyak** [comment]
> What specifically is not restored?

`./some/path` (see step 2 in "Steps to Reproduce").

**Comment by bpasero** [comment]
Ah, this is about the integrated terminal.

**Comment by Tyriar** [comment]
> "Session contents restored" works only for last closed instance of VS Code. I think this is unexpected and inconsistent behavior.

The one caveat I'm aware of here is that it will only restore the terminal if anything was input into it to avoid storing the buffer of and re-creating terminals that didn't matter. Is this the case you're hitting? Can you create a [gif](http://gifcap.dev/) showing the problem?

**Comment by KostyaTretyak** [comment]
> The one caveat I'm aware of here is that it will only restore the terminal if anything was input into it to avoid storing the buffer of and re-creating terminals that didn't matter. Is this the case you're hitting?

No, see GIF.

![vs-code-restoring](https://user-images.githubusercontent.com/2286306/153879918-533c8644-6f1d-4a50-a166-4a69b6f5b96c.gif)

**Comment by KostyaTretyak** [comment]
@Tyriar, I update GIF (added comments to it).

**Comment by KostyaTretyak** [comment]
Added label `question` and close? Hm... But default behavior is really illogical. This is from the category of "bug or feature"...
